mirascope/core/gemini/_call.py

```diff
@@ -22,10 +22,10 @@
     TToolType=GeminiTool,
     TCallParams=GeminiCallParams,
     default_call_params=GeminiCallParams(),
-    setup_call=setup_call,  # type: ignore
+    setup_call=setup_call,
     get_json_output=get_json_output,
-    handle_stream=handle_stream,
-    handle_stream_async=handle_stream_async,
+    handle_stream=handle_stream,  # pyright: ignore [reportArgumentType]
```

Comment:
Removing the type ignore in setup_call caused type checking by Generics to be performed. it is difficult to fix because the Response is different for different providers.

Reply:
but shouldn't the generics here specifically match? i.e. generics from gemini setup_call match handle_stream etc.?

Reply:
The issue here is the varying "combinations" of responses from providers.
Some return the same response for both `stream=False` and `stream=True`, while others don't. Additionally, there are various patterns regarding whether it's an `AsyncGenerator` or an `Awaitable[Stream]`. With the current function definition, it's too challenging to express all these patterns in type annotations. If possible, it would be ideal to create a new function that wraps everything, but I don't think it's worth investing that much time and effort into it.

Reply:
ok, we can revisit this later if we think it's worth the time.